### PR TITLE
add march_input_device tutorial and overview

### PR DESCRIPTION
**Pull request by TimBuckers**

### CODE REVIEW

File: doc/march_packages/march_input_device.rst [hunk 1]
```diff
@@ -0,0 +1,119 @@
+march_input_device
+==================
+
+Overview
+--------
+The march_input_device is the software running on the |m4| exoskeleton input device. The input device is used by the
+to give input to the software running on the |m4| exoskeleton.
```
Comment (Ishadijcks) [inline doc/march_packages/march_input_device.rst:7]: By the what?

File: doc/march_packages/march_input_device.rst [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+
+
+Communication
+-------------
+The communication between the input device and the |m4| exoskeleton is done with `rosserial <http://wiki.ros.org/rosserial>`_.
+This can be run wired via usb cable or wireless via wifi connection.
+
+Sending messages
```
Comment (Ishadijcks) [inline doc/march_packages/march_input_device.rst:15]: Link to the message files

File: doc/march_packages/march_input_device.rst [hunk 3]
```diff
@@ -0,0 +1,119 @@
...
+^^^^^^^^^^^^^^^^
+**Gait instruction:** This is a instruction to start the sent gait right now, if this is possible.
+
+**Stop:** This message can stop a looping gait(such as walk). After receiving the stop message the |m4| exoskeleton will go to the standing pose.
```
Comment (Ishadijcks) [inline doc/march_packages/march_input_device.rst]:
```suggestion
**Stop:** This message can stop a repeating gait (such as walk). After receiving the stop message the exoskeleton will go to the standing pose.
```

File: doc/march_packages/march_input_device.rst [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+
+.. todo:: (Tim) link this to the march_safety documentation.
+
+**Stay alive:** Every cycle a stay alive message is send. This way *march_safety* can detect when the input device loses connection.
```
Comment (Ishadijcks) [inline doc/march_packages/march_input_device.rst]: What cycle?

File: doc/march_packages/march_input_device.rst [hunk 5]
```diff
@@ -0,0 +1,119 @@
...
+
+Receiving messages
+^^^^^^^^^^^^^^^^^^
+**Gait instruction response:** This message indicate that the last gait instruction is handled. The value represent successful handled or rejected.
```
Comment (Ishadijcks) [inline doc/march_packages/march_input_device.rst]: indicates*
represents*

What does handled mean?

File: doc/march_packages/march_input_device.rst [hunk 6]
```diff
@@ -0,0 +1,119 @@
...
+- Comment :code:`#define USE_WIRELESS` in the main.cpp.
+- Upload to input device.
+
+Start the software
+~~~~~~~~~~~~~~~~~~
+- Type in a terminal:
+
+.. code::
+
+    roscore
```
Comment (Ishadijcks) [inline doc/march_packages/march_input_device.rst]: Why start a core separately?

File: doc/march_packages/march_input_device.rst [hunk 7]
```diff
@@ -0,0 +1,119 @@
...
+
+- Connect the input device via usb cable, this powers and launches the input device automatically.
+
+
+How to run Wireless
+^^^^^^^^^^^^^^^^^^^
+
+.. todo:: (Electro) Document how to change the input device to wireless mode
+
+.. note:: Make sure the pc that needs to receive the messages and the input device are on the same hotspot.
```
Comment (Ishadijcks) [inline doc/march_packages/march_input_device.rst]:
```suggestion
.. note:: Make sure the pc that needs to receive the messages and the input device are on the same network.
```

With a router it's technically not a hotspot. Message is clear though

File: doc/march_packages/march_input_device.rst [hunk 8]
```diff
@@ -0,0 +1,119 @@
...
+
+Upload the code on the input device
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Skip this section if the correct code for running wired is already on the input device. More detailed information on how
+to run and upload the code see the :input-device:`ReadMe <>`.
+
+.. note:: If you want to change the hotspot name and password that the input device uses: Change **ssid** and **password** in the *WirelessConnection.h*
```
Comment (Ishadijcks) [inline doc/march_packages/march_input_device.rst]: Once again not hotspot

File: doc/march_packages/march_input_device.rst [hunk 9]
```diff
@@ -0,0 +1,119 @@
...
+
+Sending messages
+^^^^^^^^^^^^^^^^
+**Gait instruction:** This is a instruction to start the sent gait right now, if this is possible.
+
+**Stop:** This message can stop a repeating gait (such as walk). After receiving the stop message the exoskeleton will go to the standing pose.
+
+.. todo:: (Tim) link this to the march_safety documentation.
+
+**Stay alive:** Every cycle a stay alive message is send. This way *march_safety* can detect when the input device loses connection.
```
Comment (TimBuckers) [inline doc/march_packages/march_input_device.rst]:
```suggestion
**Stay alive:** Every loop a stay alive message is send. This way *march_safety* can detect when the input device loses connection.
```

File: doc/march_packages/march_input_device.rst [hunk 10]
```diff
@@ -0,0 +1,119 @@
...
+
+**Stop:** This message can stop a repeating gait (such as walk). After receiving the stop message the exoskeleton will go to the standing pose.
+
+.. todo:: (Tim) link this to the march_safety documentation.
+
+**Stay alive:** Every loop a stay alive message is send. This way *march_safety* can detect when the input device loses connection.
+
+Receiving messages
+^^^^^^^^^^^^^^^^^^
+**Gait instruction response:** This message indicate that the last gait instruction is handled. The value represent successful handled or rejected.
```
Comment (TimBuckers) [inline doc/march_packages/march_input_device.rst]:
```suggestion
**Gait instruction response:** This message indicates that the last gait instruction is received. The value tells you if the instruction is successful handled or rejected.
```

Comment (TimBuckers) [reply doc/march_packages/march_input_device.rst]: Is this better?

File: doc/march_packages/march_input_device.rst [hunk 11]
```diff
@@ -0,0 +1,119 @@
...
+.. todo:: (Electro) Document how to change the input device to wireless mode
+
+.. note:: Make sure the pc that needs to receive the messages and the input device are on the same network.
+
+Upload the code on the input device
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Skip this section if the correct code for running wired is already on the input device. More detailed information on how
+to run and upload the code see the :input-device:`ReadMe <>`.
+
+.. note:: If you want to change the hotspot name and password that the input device uses: Change **ssid** and **password** in the *WirelessConnection.h*
```
Comment (TimBuckers) [inline doc/march_packages/march_input_device.rst]:
```suggestion
.. note:: If you want to change the network name and password that the input device uses: Change **ssid** and **password** in the *WirelessConnection.h*
```

File: doc/march_packages/march_input_device.rst [hunk 12]
```diff
@@ -0,0 +1,119 @@
+march_input_device
+==================
+
+Overview
+--------
+The march_input_device is the software running on the |m4| exoskeleton input device. The input device is used by the
```
Comment (TimBuckers) [inline doc/march_packages/march_input_device.rst]:
```suggestion
The march_input_device is the software running on the |m4| exoskeleton input device. The input device is used
```

File: doc/march_packages/march_input_device.rst (hunk 2 above)
Comment (TimBuckers) [reply doc/march_packages/march_input_device.rst:15]: done

File: doc/march_packages/march_input_device.rst (hunk 1 above)
Comment (TimBuckers) [reply doc/march_packages/march_input_device.rst:7]: fixed

**Review by Ishadijcks — APPROVED** [review]
LGTM

**Pull request merged**